pgdump-is-not-a-backup-tool.mdx

```diff
@@ -0,0 +1,70 @@
+
+Some users inadvertently rely on `pg_dump` as their primary backup tool, when better solutions exist. The `pg_dump` man page is arguably misleading in that it starts with:
+
+> "pg_dump is a utility for backing up a PostgreSQL database."
+
+This tones it down a little bit by replacing most uses of "backup" with "export" and adding a short note that `pg_dump` is not a general-purpose backup tool.
+
+**Discussion:** [https://www.postgresql.org/message-id/flat/70b48475-7706-4268-990d-fd522b038d96%40eisentraut.org](https://www.postgresql.org/message-id/flat/70b48475-7706-4268-990d-fd522b038d96%40eisentraut.org)
+
+**Commit Link:** [https://gitlab.com/postgres/postgres/-/commit/4f29394ea941f688fd4faf7260d2c198931ca797](https://gitlab.com/postgres/postgres/-/commit/4f29394ea941f688fd4faf7260d2c198931ca797)
+
+I am pleased that we are using better wording to explain the scope of `pg_dump` in the Postgres documentation.
+
+What `pg_dump` does is take a snapshot of the database at that moment. However, it excludes certain crucial components, like WAL files and the global objects in the entire Postgres cluster.
+
+In a moment of database disruption, you will need to restore your backups. And if you only have a `pg_dump` export and did not also back up your WAL directory, you would lose some of the transaction history. If you are doing replication, that would complicate things and could result in longer downtime or a more extended maintenance window if it’s a planned operation.
+
+Backups ideally should be continuous. You shouldn’t have to worry about managing WALs and other objects back and forth between servers. You should have regular backups, continuously adding to them, and ideally restore them regularly to ensure everything is going well. It’s also important to monitor their health.
+
+I won’t go into details about different backup tools here, but suffice it to say, `pg_dump` is not sufficient alone as a backup mechanism. It is a useful utility and helps a lot, but it’s not something you should consider a complete solution for backing up your Postgres clusters.
+
+### Conclusion
+
+To recap, let’s discuss a few shortcomings of `pg_dump` and clarify why a **dump** and a **backup** are different things.
+
+`pg_dump` creates a snapshot of the database at a specific point in time. This means it captures the state of the database up to that moment but does not account for any changes made afterward. The standard `pg_dump` output is a plain-text file containing SQL statements to recreate the database schema and its contents. While it can be in various formats, SQL is the most common. This output is what we refer to as a **dump**.
+
+However, this dump file does not include Write-Ahead Logs (WAL), which are essential for point-in-time recovery and replication. Additionally, global objects such as roles and tablespaces are excluded from the dump, which can lead to inconsistencies during restoration, especially in multi-database setups.
```

Comment:
I would instead reference the full "Write-Ahead Logs (WAL)" reference as the first mention of the WAL acronym in the blog.

Reply:
Ah, I replaced them all as WAL.